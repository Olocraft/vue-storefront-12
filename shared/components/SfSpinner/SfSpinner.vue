<template>
  <button>Button</button>
</template>

<script>
export default {
  name: 'SfSpinner'
}
</script>
