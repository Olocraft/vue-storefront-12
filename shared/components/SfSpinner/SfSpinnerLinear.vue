<template>
  <div class="relative w-10 h-1 overflow-hidden">
    <div class="absolute w-10 h-1 bg-gray-200"></div>
    <div class="absolute w-2.5 h-1 animate-spin-l bg-primary-600"></div>
  </div>
</template>

<script>
export default {
  name: 'SfSpinnerLinear'
}
</script>
