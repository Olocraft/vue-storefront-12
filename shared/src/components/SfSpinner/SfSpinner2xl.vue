<template>
  <div
    class="relative h-24"
    role="presentation"
    aria-live="polite"
    aria-busy="true"
  >
    <div class="absolute w-24 h-24 bg-transparent border-4 border-gray-200 rounded-full" />
    <div class="absolute w-12 h-12 bg-transparent border-4 border-transparent rounded-tl-full stopped-animation animate-spin border-t-primary-600" />
  </div>
</template>

<script>
export default {
  name: 'SfSpinner2xl'
}
</script>
