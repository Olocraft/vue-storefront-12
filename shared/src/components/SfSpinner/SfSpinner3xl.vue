<template>
  <div
    class="relative h-48"
    role="presentation"
    aria-live="polite"
    aria-busy="true"
  >
    <div class="absolute w-48 h-48 bg-transparent border-8 border-gray-200 rounded-[100%]" />
    <div class="stopped-animation absolute w-24 h-24 bg-transparent border-8 border-transparent rounded-tl-[9999px] animate-spin border-t-primary-600" />
  </div>
</template>

<script>
export default {
  name: 'SfSpinner3xl'
}
</script>
