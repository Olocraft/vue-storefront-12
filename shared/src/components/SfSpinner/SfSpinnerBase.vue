<template>
  <div
    class="relative h-6"
  >
    <div
      class="sr-only"
      role="alert"
    >
      loading...
    </div>
    <div class="absolute w-6 h-6 bg-transparent border-2 border-gray-200 rounded-full" />
    <div class="w-3 h-3 bg-transparent border-2 border-transparent rounded-tl-full stopped-animation before:absolute animate-spin border-t-primary-600" />
  </div> 
</template>

<script>
export default {
  name: 'SfSpinnerBase'
}
</script>
