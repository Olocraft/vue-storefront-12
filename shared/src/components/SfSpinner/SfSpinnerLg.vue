<template>
  <div class="relative">
      <div class="absolute w-10 h-10 bg-transparent border-2 border-gray-200 rounded-full"></div>
      <div class="absolute w-5 h-5 bg-transparent border-2 border-transparent rounded-tl-full animate-spin border-t-primary-600"></div>
    </div>
</template>

<script>
export default {
  name: 'SfSpinnerLg'
}
</script>
