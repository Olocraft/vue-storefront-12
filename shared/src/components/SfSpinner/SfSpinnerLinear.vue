<template>
  <div
    class="relative w-10 h-1 overflow-hidden"
    role="presentation"
    aria-live="polite"
    aria-busy="true"
  >
    <div class="absolute w-10 h-1 bg-gray-200" />
    <div class="absolute w-2.5 h-1 animate-linear-loader bg-primary-600" />
  </div>
</template>

<script>
export default {
  name: 'SfSpinnerLinear'
}
</script>
