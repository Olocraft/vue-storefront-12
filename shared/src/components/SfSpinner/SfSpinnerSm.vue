<template>
  <div
    class="relative h-4"
  >
    <div
      class="sr-only"
      role="alert"
    >
      loading...
    </div>
    <div class="absolute w-4 h-4 bg-transparent border-2 border-gray-200 rounded-3xl" />
    <div class="absolute w-[10px] h-[10px] bg-transparent border-2 border-transparent rounded-tl-xl animate-spin border-t-primary-600" />
  </div>
</template>

<script>
export default {
  name: 'SfSpinnerSm'
}
</script>
