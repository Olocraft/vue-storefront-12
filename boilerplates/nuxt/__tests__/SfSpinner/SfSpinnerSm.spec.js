import { shallowMount } from '@vue/test-utils'
import SfSpinner from '../../components/SfSpinner/SfSpinnerSm.vue'
let wrapper

describe('SfSpinnerSm.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(SfSpinner)
  })
  it('render component correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
