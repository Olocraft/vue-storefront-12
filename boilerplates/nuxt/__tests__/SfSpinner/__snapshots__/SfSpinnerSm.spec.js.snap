// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SfSpinnerSm.vue render component correctly 1`] = `
<div role="presentation" aria-live="polite" aria-busy="true" class="relative h-4">
  <div class="absolute w-4 h-4 bg-transparent border-2 border-gray-200 rounded-3xl"></div>
  <div class="absolute w-[10px] h-[10px] bg-transparent border-2 border-transparent rounded-tl-xl animate-spin border-t-primary-600"></div>
</div>
`;
